virtcontainers/mount_test.go

```diff
@@ -344,6 +344,9 @@ func TestIsEphemeralStorage(t *testing.T) {
 // or directory attempting to be unmounted doesn't exist, then it
 // is not considered an error
 func TestBindUnmountContainerRootfsENOENTNotError(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("Test disabled as requires root user")
+	}
```

Comment:
This could be the more general:

```go
if tc.NotValid(ktu.NeedRoot()) {
    t.Skip(ktu.TestDisabledNeedRoot)         
}
```

___

Thought: In fact, we could simplify the constraints API to call `t.Skip()` itself:

```go
func (tc *TestConstraint) SkipIfNotValid(t *testing.T, constraints ...Constraint) {
    if tc.NotValid(t, constraints...) {
        t.Skip()
   }
}
```

But that would require a bit more rework to auto-generate the message passed to `t.Skip()`.